src/sidebar/test/frame-sync-test.js

```diff
@@ -189,16 +189,39 @@ describe('FrameSync', function () {
   });
 
   context('when anchoring completes', function () {
+    var clock = sinon.stub();
+
+    beforeEach(() => {
+      clock = sinon.useFakeTimers();
+    });
+
+    afterEach(() => {
+      clock.restore();
+    });
+
     it('updates the anchoring status for the annotation', function () {
       fakeBridge.emit('sync', [{tag: 't1', msg: {$orphan: false}}]);
-      assert.calledWith(fakeAnnotationUI.updateAnchorStatus, null, 't1', false);
+      clock.tick(20);
+      assert.calledWith(fakeAnnotationUI.updateAnchorStatus, { t1: 'anchored' });
+    });
+
+    it('coalesces multiple "sync" messages', () => {
+      fakeBridge.emit('sync', [{tag: 't1', msg: {$orphan: false}}]);
+      fakeBridge.emit('sync', [{tag: 't2', msg: {$orphan: true}}]);
+      clock.tick(20);
+
+      assert.calledWith(fakeAnnotationUI.updateAnchorStatus, {
+        t1: 'anchored',
+        t2: 'orphan',
+      });
     });
 
     it('emits an ANNOTATIONS_SYNCED event', function () {
       var onSync = sinon.stub();
       $rootScope.$on(events.ANNOTATIONS_SYNCED, onSync);
 
       fakeBridge.emit('sync', [{tag: 't1', msg: {$orphan: false}}]);
+      clock.tick(20);
```

Comment:
This is needed because emitting the `ANNOTATIONS_SYNCED` event no longer happens synchronously, due to debouncing, when a "sync" message is received by the sidebar.